docs/1.getting-started/5.seo-meta.md

```diff
@@ -128,6 +128,29 @@ interface MetaObject {
 
 See [@unhead/schema](https://github.com/unjs/unhead/blob/main/packages/schema/src/schema.ts) for more detailed types.
 
+## Pitfalls
+
+### Async useHead / useSeoMeta
+
+A common use case when working with SEO composables is to fetch data from an API and use it to set the page title.
+However, running `useHead` after an `await` will cause the Nuxt app context to be lost.
+
+To solve this, you can either:
+
+- Use `runWithContext` to preserve to context:
+
+```ts
+const nuxtApp = useNuxtApp();
+const data = await $fetch(`/api/data/${key}`) // context is lost from the await
+nuxtApp.runWithContext(() => {
+  useHead({
+    title: `${data.title}`,
+  })
+});
```

Comment:
do you think we can create a wrapper around `useHead` and other head composables that does this automatically, instead of asking users to do this?

Reply:
How can it solve the lost context from the await though :thinking:

Reply:
To give some more context, if the vueApp is available then useHead will work. So the only time this error comes up is from edge cases like https://github.com/nuxt/nuxt/issues/19244

From my understanding, `asyncContext` is aiming to fix users having to do anything to solve it.

Reply:
asyncContext is great but currently only for node + bun.

In `<script setup>` vue transforms it and we preserve vueApp context. Elsewhere, there are places where we transform things to preserve _nuxt_ context but we can't preserve the vue context as we're not in control there. So we might have nuxtApp (from which we can access vue app) even where inject might not work.

Reply:
I might have solved this, can you review :pray:

Reply:
this looks _very_ promising. just one question for you but otherwise nice work